pkg/api/store/globaldns/globaldns_store.go

```diff
@@ -0,0 +1,86 @@
+package globaldns
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	managementv3 "github.com/rancher/types/client/management/v3"
+)
+
+func Wrap(store types.Store) types.Store {
+	storeWrapped := &Store{
+		Store: store,
+	}
+	storeWrapped.mu = sync.Mutex{}
+	return storeWrapped
+}
+
+type Store struct {
+	types.Store
+	mu sync.Mutex
+}
+
+func (p *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+	fqdn := convert.ToString(data["fqdn"])
+	if fqdn == "" {
```

Comment:
shall we make fqdn required in the norman framework?

Reply:
Looks like it's already marked as required: https://github.com/rancher/types/blob/7746d46d672f3811e95b1629db079a046dd4f8a5/apis/management.cattle.io/v3/globaldns_types.go#L21

Reply:
yeah its already required, can relax the checks